plugins/linux-v4l2/v4l2-input.c

```diff
@@ -1008,11 +1010,18 @@ static void v4l2_init(struct v4l2_data *data)
 	blog(LOG_INFO, "Linesize: %d Bytes", data->linesize);
 
 	/* set framerate */
-	if (v4l2_set_framerate(data->dev, &data->framerate) < 0) {
-		blog(LOG_ERROR, "Unable to set framerate");
-		goto fail;
+	if (data->framerate == 0) {
+		blog(LOG_ERROR, "Framerate is not set, falling back to %i",
+		     FALLBACK_FRAMERATE);
+		data->framerate = v4l2_pack_tuple(1, FALLBACK_FRAMERATE);
+		if (v4l2_set_framerate(data->dev, &data->framerate) < 0) {
```

Comment:
This should be outside of the if statement to preserve functionality for non-broken webcams.